Intro (#51)
infinum/eightshift-docs: website/docs/basics/manifest.md

## How to use it?

First, install the manifest class if you didn't use the default project installation using this command:

`wp boilerplate create_manifest`

You will see a custom filter that you can use to fetch the item from the public folder in your class. This is the custom filter:

```js
/**
 * Manifest item filter name constant.
 *
 * @var string
 */
public const MANIFEST_ITEM = 'manifest-item';

/**
 * Register all hooks. Changed filter name to manifest.
 *
 * @return void
 */
public function register(): void
{
	\add_filter(static::MANIFEST_ITEM, [ $this, 'getAssetsManifestItem' ]);
}
```

To use this filter, add this code and provide the correct name of your asset:

```js
use CoolProject\Manifest\Manifest;

$logo = \apply_filters(Manifest::MANIFEST_ITEM, 'logo.svg');
```

And that is it. This filter will check the public folder for the `manifest.json` file, parse it and return the value of the provided key.

If there is no `manifest.json` file or you provided the wrong asset name, there will be a descriptive message for you.

## Why not just fetch the asset the old fashioned way?

> By old fashioned way we mean ruining something like this:

Reviewer: ```suggestion
By old fashioned way we mean running something like this:
```

> ```js

Reviewer 2: ```suggestion
```php
```

$logo = get_template_directory_uri() . 'public/logo.svg';
```

> You definitely can do this, but with our filter, if you want to change the public folders location or public folders name for some reason, you can change it at one place, and you are done.

Reviewer 2: ```suggestion
You can definitely do this. But with our filter, if you want to change the public folder location or public folder name for some reason, you can change it in one place, and you are done.
```

> If you are using the old fashioned way, you would need to search and replace the whole project and implement the change. There is always a chance you would miss something, break the project, get yelled at by your boss, or get fired.

Reviewer 2: ```suggestion
If you are using the old fashioned way, you would need to search and replace the whole project and implement the change. There is always a chance you would miss something, break the project, etc.
```

I really hope that people don't fire other people for such mistakes 👀

> And this is the modern way, so there is that.

Reviewer 2: Yeah, I'd leave this one out. Just because something is modern or cool doesn't mean we need to use it 😂
